importer/loader.py

```diff
@@ -83,6 +90,86 @@ def load(self):
+                result.limit = self._sheet.cell(row=i + 1, column=2).value.strftime('%d/%m/%Y')
+            if self._sheet.cell(row=i + 1, column=1).value == 'Filename' or self._sheet.cell(row=i + 1, column=1).value == 'Run Accession':
+                data_row = i + 2
+                header_row = i + 1
+                break
+        filename_column = None
+        run_accession_column = None
+        for i in range(self._sheet.max_column):
+            if self._sheet.cell(row=header_row, column=i + 1).value == 'Filename':
+                filename_column = i +1
+            if self._sheet.cell(row=header_row, column=i + 1).value == 'Run Accession':
+                run_accession_column = i +1
+            if filename_column is not None:
+                if self._sheet.cell(row=header_row, column=i + 1).value == 'Mate File':
+                    mate_filename_column = i + 1
+            if run_accession_column is not None:
+                if self._sheet.cell(row=header_row, column=i + 1).value == 'Double-ended Reads':
+                    double_ended_reads_column = i + 1
+            if self._sheet.cell(row=header_row, column=i + 1).value == 'Sample Name':
+                sample_name_column = i + 1
+            if self._sheet.cell(row=header_row, column=i + 1).value == 'Sample Accession number':
+                sample_accession_column = i + 1
+            if self._sheet.cell(row=header_row, column=i + 1).value == 'Taxon ID':
+                taxon_id_column = i + 1
+            if self._sheet.cell(row=header_row, column=i + 1).value == 'Library Name':
+                library_name_column = i + 1
+        reads = []
+        for i in range(data_row, self._sheet.max_row+1):
+            sample_name = self.__extract_float_value_xlsx(i, sample_name_column)
+            library_name = self.__extract_float_value_xlsx(i, library_name_column)
+            if library_name is None:
+                library_name = sample_name
+            if filename_column is not None:
+                reads.append(RawRead(
+                    self.__extract_text_value_xlsx(i, filename_column),
+                    self.__extract_text_value_xlsx(i, mate_filename_column),
+                    sample_name,
+                    self.__extract_text_value_xlsx(i, sample_accession_column),
+                    self.__extract_float_value_xlsx(i, taxon_id_column),
+                    library_name))
+            if run_accession_column is not None:
+                reads.append(RawRead(
+                    (self.__extract_text_value_xlsx(i, run_accession_column)),
+                    self.__extract_text_value_xlsx(i, double_ended_reads_column),
+                    sample_name,
+                    self.__extract_text_value_xlsx(i, sample_accession_column),
+                    self.__extract_float_value_xlsx(i, taxon_id_column),
+                    library_name))
+        result.reads = reads
+        return result
```

Comment:
There is quite a lot of repeated code in this and the function above. Could this be extracted?